### PR TITLE
caps: autoLaunch option

### PR DESCRIPTION
**Pull request by scottdixon**

### CODE REVIEW

**Comment by scottdixon** [comment]
@imurchie @moizjv

**Comment by scottdixon** [comment]
@moiz - caps seems to be consistent for most tests - we could move these out, any objections?

File: test/functional/driver-e2e-specs.js [hunk 1]
```diff
@@ -66,6 +62,30 @@ describe('createSession', function () {
     };
     await driver.createSession(caps).should.eventually.be.rejectedWith(/Could not find app/);
   });
+  xit('should be able to start session without launching app', async () => {
```
Comment (moizjv) [inline test/functional/driver-e2e-specs.js]: xit?

File: test/functional/driver-e2e-specs.js [hunk 2]
```diff
@@ -66,6 +62,30 @@ describe('createSession', function () {
...
+      deviceName: 'Android',
+      platformName: 'Android',
+      appPackage: pkg,
+      appActivity: act,
+      autoLaunch: false
+    };
+    await driver.createSession(caps);
+    await driver.adb.getFocusedPackageAndActivity().should.eventually.be.null;
+  });
+  it('should be able to start session without installing app', async () => {
```
Comment (moizjv) [inline test/functional/driver-e2e-specs.js]: Can we club this with above test?

File: lib/driver.js [hunk 3]
```diff
@@ -128,7 +129,7 @@ class AndroidDriver extends BaseDriver {
     if (this.opts.ignoreUnimportantViews) {
       await this.settings.update({ignoreUnimportantViews: this.opts.ignoreUnimportantViews});
     }
-    if (!this.appOnDevice) {
+    if (!this.appOnDevice && this.opts.autoLaunch) {
```
Comment (moizjv) [inline lib/driver.js]: Can we add a comment here about autoLaunch, that user is responsible for init and startapp?

File: test/functional/driver-e2e-specs.js (hunk 1 above)
Comment (imurchie) [inline test/functional/driver-e2e-specs.js]: same as `it.skip`.

File: test/functional/driver-e2e-specs.js [hunk 4]
```diff
@@ -3,69 +3,52 @@ import chaiAsPromised from 'chai-as-promised';
...
-    let pkg = 'io.appium.android.apis';
-    let act = '.view.SplitTouchView';
-    let caps = {
-      app: "foo",
-      deviceName: 'Android',
-      platformName: 'Android',
-      appPackage: pkg,
-      appActivity: act
-    };
+    let caps = Object.assign({}, defaultCaps, {app: 'foo'});
```
Comment (scottdixon) [inline test/functional/driver-e2e-specs.js]: Should these be over two lines? e.g.
```let caps = Object.assign({}, defaultCaps);```
```caps.app = 'foo';```

Comment (imurchie) [inline test/functional/driver-e2e-specs.js]: This is fine.

File: test/functional/driver-e2e-specs.js [hunk 5]
```diff
@@ -3,69 +3,52 @@ import chaiAsPromised from 'chai-as-promised';
...
 import sampleApps from 'sample-apps';
-//import ADB from 'appium-adb';
-
-let apiDemos = sampleApps('ApiDemos-debug');
 
 chai.should();
 chai.use(chaiAsPromised);
 
 describe('createSession', function () {
   let driver;
```
Comment (imurchie) [inline test/functional/driver-e2e-specs.js]: I think this should just be the shared caps

```js
let defaultCaps = {
    app: sampleApps('ApiDemos-debug'),
    deviceName: 'Android',
    platformName: 'Android'
  };
```

And then add

```js
let appPackage = 'io.appium.android.apis';
let appActivity = '.view.SplitTouchView';
```

**Comment by moizjv** [comment]
LGTM

**Pull request merged**